activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -163,6 +163,25 @@ def load_records(raw_records = nil)
           end
         end
 
+        def associate_records_from_unscoped(records)
+          return unless records
```

Comment:
Should this be `return if records.empty?` ?

Reply:
It should guard against nil and empty arrays, will update and push!